topics/go/algorithms/strings/reverse/reverse.go

```diff
@@ -0,0 +1,30 @@
+package strings
+
+func ReverseString(str string) string {
+
+	// Here we create a slice of runes.
+	runes := []rune(str)
+
+	// Here we create int that will be a pointer to the front of the runes.
+	var beg int
+
+	// Here  we create int that will be a pointer to the end of the runes.
+	end := len(runes) - 1
+
+	// While there are still code points to check.
+	for beg < end {
+
+		// Swap the code point.
+		r := runes[beg]
+
+		runes[beg] = runes[end]
+
+		runes[end] = r
+
+		beg, end = beg+1, end-1
```

Comment:
Stop doing this. This does not help readability. Break this down into two lines.
// Move the indexes closer to each other working towards the middle of the collection.

Reply:
Completed